NullReferenceException in HandlerPinsOnCollectionChanged

There are no null check in HandlerPinsOnCollectionChanged, NotifyCollectionChangedEventArgs:NewItems can be null

```
 [ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object.
   at Mapsui.UI.Forms.MapView.HandlerPinsOnCollectionChanged (System.Object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e) [0x000e2] in C:\src\Mapsui\Mapsui.UI.Forms\MapView.cs:692 
   at System.Collections.ObjectModel.ObservableCollection`1[T].OnCollectionChanged (System.Collections.Specialized.NotifyCollectionChangedEventArgs e) [0x00018] in <3e9b3e26c4694baab3f689687ad40612>:0 
   at System.Collections.ObjectModel.ObservableCollection`1[T].OnCollectionReset () [0x00000] in <3e9b3e26c4694baab3f689687ad40612>:0 
   at System.Collections.ObjectModel.ObservableCollection`1[T].ClearItems () [0x00018] in <3e9b3e26c4694baab3f689687ad40612>:0 
   at System.Collections.ObjectModel.Collection`1[T].Clear () [0x00014] in <d4a23bbd2f544c30a48c44dd622ce09f>:0 
```


I fixed it in my forked repository:
https://github.com/zbyszekpy/Mapsui/commit/206b049aabc42ae2ba21e72e15e39f2c658a8e7f

Please let me know if you want me to create PR

Step to reproduce:
 ```
Samples/Mapsui.Samples.Forms/Mapsui.Samples.Forms.Shared/PinSample.cs | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Samples/Mapsui.Samples.Forms/Mapsui.Samples.Forms.Shared/PinSample.cs b/Samples/Mapsui.Samples.Forms/Mapsui.Samples.Forms.Shared/PinSample.cs
index 4f9e37bc..66ff1ca7 100644
--- a/Samples/Mapsui.Samples.Forms/Mapsui.Samples.Forms.Shared/PinSample.cs
+++ b/Samples/Mapsui.Samples.Forms/Mapsui.Samples.Forms.Shared/PinSample.cs
@@ -23,6 +23,8 @@ namespace Mapsui.Samples.Forms
             var mapView = sender as MapView;
             var e = args as MapClickedEventArgs;
 
+            mapView.Pins.Clear();
+
             var assembly = typeof(MainPageLarge).GetTypeInfo().Assembly;
             foreach (var str in assembly.GetManifestResourceNames())
                 System.Diagnostics.Debug.WriteLine(str);

```
Br
Zbyszek

--- a/Mapsui.UI.Forms/MapView.cs
+++ b/Mapsui.UI.Forms/MapView.cs
@@ -689,12 +689,15 @@ namespace Mapsui.UI.Forms
                 }
             }
 
-            foreach (var item in e.NewItems)
+            if (e.NewItems != null)
             {
+              foreach (var item in e.NewItems)
+              {
                 // Add new pins to layer
                 var pin = item as Pin;
 
                 pin.PropertyChanged += HandlerPinPropertyChanged;
+              }
             }
 
             Refresh();

